### PR TITLE
Closes #712 Data type comparison

### LINKED ISSUE #712: bids_FullPipelineTest: Nifti comparisons have unequal data format
### Description

Nifti comparisons can have unequal data format. Line 107 of `xASL_im_CompareNifti` assumes that the image matrices loaded from NIfTIs have equal data format, so this crashes when this is not the case. Can be solved by checking if the data format of the images is equal first. 

This issue was raised when running `
xASL_test_Flavors(pathExploreASL, pathTest, [0 0 1 0 0 0 0], x);` for `Siemens_PCASL_multiTI`. Somehow the `ASL4D` of `TmpConversion` had `uint8` format whereas the `TmpReference` `ASL4D` had `single` format.

I recently changed that `xASL_io_SaveNifti` automatically stores in lower data formats if this doesn't affect the data quality, and checked but this was not an issue (or at least I couldn't reproduce this by storing this image as new NIfTI). So it seems that my change (merged in 1.7.0) was appropriate.

### Release notes

Required: summarize the changes for the release notes here

### PR DESCRIPTION
**Pull request by MichaelStritt**
### Linked issue

Check out #712

### How to test

Required: if not defined in the linked issue, add a simple test description here

### Comments

Does this work for you?

### CODE REVIEW

File: Functions/xASL_im_CompareNiftis.m [hunk 1]
```diff
@@ -67,10 +67,17 @@
...
-    sizeB = structB.bytes;
+    if ~isempty(structA) && ~isempty(structB)
+        sizeA = structA.bytes;
+        sizeB = structB.bytes;
+    else
+        warning('Problem loading the images...');
+        identical = false;
+        return
+    end
     if ~(sizeA>1) || ~(sizeB>1)
```
Comment (jan-petr) [inline Functions/xASL_im_CompareNiftis.m:78]: what about (sizeA<=1) ;)

**Review by jan-petr — CHANGES_REQUESTED** [review]
Minor things. Otherwise good.

File: Functions/xASL_im_CompareNiftis.m (hunk 1 above)
Comment (MichaelStritt) [reply Functions/xASL_im_CompareNiftis.m]: Fixed

**Comment by MichaelStritt** [comment]
### Unit test

```
>> UnitTest = xASL_ut_function_xASL_im_CompareNiftis(TestRepository);
Identical: true
RMSE:      0.00
Min.diff.: 0.00
Max.diff.: 0.00
Identical: false
RMSE:      4.45
Min.diff.: 0.00
Max.diff.: 57.00
```

**Review by jan-petr — APPROVED** [review]
OK.

**Pull request merged**